cocotb/scheduler.py

```diff
@@ -547,12 +547,18 @@ def _coroutine_yielded(self, coro, trigger):
                 self._trigger2coros.pop(trigger)
 
                 # replace it with a new trigger that throws back the exception
-                error_trigger = NullTrigger(outcome=outcomes.Error(e))
-                self._coro2trigger[coro] = error_trigger
-                self._trigger2coros[error_trigger] = [coro]
+                self._trigger_task_with_outcome(coro, trigger_name="Trigger.prime() Error", outcome=outcomes.Error(e))
+
+    def _trigger_task_with_outcome(self, task, trigger_name=None, outcome=None):
+        """Trigger task with new outcome"""
+        new_trigger = NullTrigger(name=trigger_name, outcome=outcome)
+
+        self._coro2trigger[task] = new_trigger
+        self._trigger2coros[new_trigger] = [task]
+
+        # wake up the Task
+        new_trigger.prime(self.react)
```

Comment:
Why not just:
```suggestion
        self._coroutine_yielded(task, new_trigger)
```
optimization doesn't seem particularly worthwhile here.

Reply:
Perhaps `_coroutine_yielded(coro, trigger)` should be renamed something like `_resume_coro_upon(coro, trigger)`.

Reply:
If he does that he could just inline the method as well.